### DISCUSSION TITLE
Documentation as a PDF

### DISCUSSION THREAD

**Discussion by brownp2k** [snl-dakota/dakota#45] [Q&A] [answered]
Unless I'm missing something, it seems the documentation PDF file(s) are no longer available which makes it impossible to view documentation for previous versions or download it for offline reference.

Is the PDF available somewhere that I missed? If not, can it please be produced as a release artifact again?

**Comment by richardsonwk** [comment]
I attempted to follow the instructions at https://snl-dakota.github.io/docs/6.18.0/users/developingdakota/writingdocs.html to build the documentation from source. The HTML built, but I was unable to get the `docs-pdf` project to produce anything, and I'm not CMake/VS-savvy enough to figure out why. The `docs-dev-pdf` did a lot of stuff but failed with `System cannot find the specified file`.

Attempting to generate PDFs manually with sphinx and latexmk resulted in hundreds of of LaTeX errors similar to the following:

```
> sphinx-build -b latex . <target dir>
...
lots of warnings about missing include files and undefined labels
...

> latexmk -interaction=nonstopmode -pdflatex=lualatex -f -pdf
...
l.728 \end{eqnarray}\end{split}
                             
You have given more \span or & marks than there were
in the preamble to the \halign or \valign now in progress.
So I'll assume that you meant to type \cr instead.
...
(That makes 100 errors; please try again.)
...
!  ==> Fatal error occurred, no output PDF file produced!
```

Maybe worth noting that using the default `pdftex` instead of `lualatex` causes the process to run out of memory (fixable with `-extra-mem-bot` but then results in the same errors as above):

```
! TeX capacity exceeded, sorry [main memory size=5000000].
\sphinxafterbreak ->\copy \sphinxcontinuationbox

l.13578           |

If you really absolutely need more capacity,
you can ask a wizard to enlarge me.
```

**Comment by briadam** [comment]
@richardsonwk alas, the Dakota team has not been able to make a PDF of the user manual either. I started digging in on the issues, but couldn't justify putting higher priority on it. 

@brownp2k For Dakota 6.16 and earlier you can download a .tar.gz of the reference and PDF of the user and theory manuals here: https://dakota.sandia.gov/manuals/. For the 6.17 and newer Sphinx manual, I expect we could similarly provide a .tar.gz and/or .zip archive of the HTML user manual.

Would those meet your needs or is PDF crucial?

**Comment by brownp2k** [comment]
> @brownp2k For Dakota 6.16 and earlier you can download a .tar.gz of the reference and PDF of the user and theory manuals here: https://dakota.sandia.gov/manuals/. For the 6.17 and newer Sphinx manual, I expect we could similarly provide a .tar.gz and/or .zip archive of the HTML user manual.
> 
> Would those meet your needs or is PDF crucial?

The HTML bundle should be fine for our needs, although a PDF would still be helpful/useful.

**Comment by brownp2k** [comment]
> For the 6.17 and newer Sphinx manual, I expect we could similarly provide a .tar.gz and/or .zip archive of the HTML user manual.

@briadam Any update on the availability of this?

**Reply by jadamstephens** [reply, marked as answer]
I just added links at https://snl-dakota.github.io

**Reply by brownp2k** [reply]
Fantastic, thank you.
